lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -116,6 +124,80 @@ function preflightCheck(
     }
 }
 
+/**
+ * Public method to indicate whether platform broker is available to make native token request.
+ */
+export async function isPlatformBrokerAvailable(
+    logger?: LoggerOptions,
+    performanceClient?: IPerformanceClient
+): Promise<boolean> {
+    const defaultLoggerOptions: LoggerOptions = {
+        loggerCallback: (): void => {
+            // Empty logger callback
+        },
+        piiLoggingEnabled: false,
+        logLevel: LogLevel.Trace,
+    };
+
+    const temporaryLogger = new Logger(
+        logger || defaultLoggerOptions,
+        name,
+        version
+    ); // Default logger
+
+    const defaultPerformanceClientConfig = {
+        auth: {
+            clientId: "test-clientid",
+        },
+    };
+    const temporaryTelemetryClient: IPerformanceClient =
+        performanceClient ||
+        new BrowserPerformanceClient(defaultPerformanceClientConfig);
+
+    temporaryLogger.trace("isPlatformBrokerAvailable called");
+
+    // Check if DOM platform API is supported
+
+    // @ts-ignore
+    if (window.navigator?.platformAuthentication) {
```

Comment:
We should also check that window is defined and return false if not

Reply:
Yes, may be also a trace statement. Do we identify non-DOM environments today in logs?